File: Week1/하동헌/큐 2.py
import sys
from collections import deque

N = int(input())

queue = deque()
result = []
for _ in range(N):
    # N의 값이 크기 때문에 반복문 O(N)동안 input() 으로 입력받으면 시간초과가 난다
    cmd = sys.stdin.readline().split() 

    if cmd[0] == "push":
        queue.append(cmd[1])
    elif cmd[0] == "pop":
        if queue:
            result.append(queue.popleft())
        else:
            result.append('-1')
    elif cmd[0] == "size":
        result.append(str(len(queue)))
    elif cmd[0] == "empty":
        if queue:
            result.append('0')
        else:
            result.append('1')
    elif cmd[0] == "front":
        if queue:
            result.append(queue[0])
        else:
            result.append('-1')
    elif cmd[0] == "back":
        if queue:
            result.append(queue[-1])
        else:
            result.append('-1')

print("\n".join(result))

File: Week1/장수희/큐2.js
const fs = require("fs");
const input = fs.readFileSync(0, "utf-8").trim().split("\n");

const N = Number(input[0]);

let queue = new Array(N);
let head = 0;
let tail = 0;
let result = [];

for (let i = 1; i <= N; i++) {
  const temp = input[i];
  const [command, value] = temp.split(" ");

  if (command === "push") {
    queue[tail++] = Number(value);
  } else if (command === "pop") {
    result.push(head === tail ? -1 : queue[head++]);
  } else if (command === "size") {
    result.push(tail - head);
  } else if (command === "empty") {
    result.push(head === tail ? 1 : 0);
  } else if (command === "front") {
    result.push(head === tail ? -1 : queue[head]);
  } else if (command === "back") {
    result.push(head === tail ? -1 : queue[tail - 1]);
  }
}

console.log(result.join("\n"));